internal/provider/resource_integration.go

```diff
@@ -411,8 +411,24 @@ func (r *integrationResource) Delete(ctx context.Context, req resource.DeleteReq
 		return
 	}
 
-	//If one of the endpoints is an offer then we need to remove the remote offer rather than destroying the integration
+	var removeRemoteOffer bool = false
+	//If one of the endpoints is an offer then we need to remove the remote offer if there is one consumer
 	if offer != nil {
+		offer_response, err  := r.client.Offers.ReadOffer(&juju.ReadOfferInput{
+			OfferURL: *offer,
+		})
+		if err != nil {
+			resp.Diagnostics.AddError("Client Error", err.Error())
+			return
+		}
+
+		// If there is only one connection in the offer, remove the remote offer
+		if offer_response.ConnectionsCount == 1 {
+			removeRemoteOffer = true
+		}
+	}
+
+	if removeRemoteOffer {
```

Comment:
I tested some cases with removeRemoteOffer and it is difficult to determine on which condition/logic RemoveRemoteOffer should be called. 
This is because
* When 2 application consumer inegrations to same offer is removed, the terraform runs the actions in parallel and so ReadOffer shows 2 connections.
* Tried to run terraform apply with -parallelism 1 and still the ReadOffer shows 2 connections for both the operations in this function.

I am thinking the consequences on discarding RemoveRemoteOffer on last connection removal. The entry will be in database but removing the model or application (without --force) using juju cli cleans up the resources without any errors.
Applying terraform plan by re-adding the integrations and removing them multiple times did not result in any errors too.

So what do you think of discarding RemoveRemoteOffer step given not sure of when to trigger this function and no consequences of leaving stale record.

Your thoughts please?
(For now i have reverted back the PR to the previous solution and wait for your comments/thoughts)

Reply:
The connection count must not go down fast enough for us to catch it when removing the integration. Nor is it worth adding a wait in this scenario. As you say, it's okay to leave the SaaS around, it may be used later.